File: src/main/java/me/kingtux/kingserverinfo/events/ClickEvent.java
package me.kingtux.kingserverinfo.events;

import me.kingtux.kingserverinfo.KingServerInfoMain;
import me.kingtux.kingserverinfo.mediagui.MediaGuiItem;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;

public class ClickEvent implements Listener {

    private KingServerInfoMain plugin;

    public ClickEvent(KingServerInfoMain pl) {
        plugin = pl;
    }

    @EventHandler
    public void playerClickEvent(InventoryClickEvent e) {
        if (e.getClickedInventory().equals(plugin.getMediaGui().getMediaGui())) {
            for (MediaGuiItem item : plugin.getMediaGui().getguiItems().keySet()) {
                if (e.getCurrentItem().isSimilar(plugin.getMediaGui().getguiItems().get(item))) {
                    if (item.isClickable()) {
                        e.getWhoClicked().spigot().sendMessage(item.getLink());
                        e.getWhoClicked().closeInventory();
                        e.setCancelled(true);
                        break;
                    } else {
                        e.setCancelled(true);
                        break;
                    }

                }
            }


        }

    }
}

File: src/me/kingtux/kingserverinfo/events/ClickEvent.kt
package me.kingtux.kingserverinfo.events

import me.kingtux.kingserverinfo.KingServerInfoMain
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent

class ClickEvent(private val plugin: KingServerInfoMain) : Listener {

    @EventHandler
    fun playerClickEvent(e: InventoryClickEvent) {
        if (e.clickedInventory == plugin.mediaGui!!.mediaGui) {
            for (item in plugin.mediaGui!!.getguiItems().keys) {
                if (e.currentItem.isSimilar(plugin.mediaGui!!.getguiItems()[item])) {
                    if (item.isClickable!!) {
                        e.whoClicked.spigot().sendMessage(item.link)
                        e.whoClicked.closeInventory()
                        e.isCancelled = true
                        break
                    } else {
                        e.isCancelled = true
                        break
                    }

                }
            }


        }

    }
}
